src/containers/pages/FocusInvestigation/map/active/index.tsx

```diff
@@ -126,21 +142,62 @@ class SingleActiveFIMap extends React.Component<
       fetchGoalsActionCreator,
       fetchJurisdictionsActionCreator,
       fetchPlansActionCreator,
+      fetchStructuresActionCreator,
       fetchTasksActionCreator,
+      plan,
     } = this.props;
-
-    await supersetFetch(SUPERSET_JURISDICTIONS_SLICE).then((result: Jurisdiction[]) =>
-      fetchJurisdictionsActionCreator(result)
-    );
-    await supersetFetch(SUPERSET_PLANS_SLICE).then((result2: Plan[]) =>
-      fetchPlansActionCreator(result2)
-    );
-    await supersetFetch(SUPERSET_GOALS_SLICE).then((result3: Goal[]) =>
-      fetchGoalsActionCreator(result3)
-    );
-    await supersetFetch(SUPERSET_STRUCTURES_SLICE, { row_limit: 3000 }).then((result4: Task[]) =>
-      fetchTasksActionCreator(result4)
-    );
+    const planId = plan && plan.plan_id;
+    if (planId) {
+      const pivotParams = superset.getFormData(3000, [
+        { comparator: planId, operator: '==', subject: 'plan_id' },
+      ]);
+      let sqlFilterExpression = '';
+      await supersetFetch(SUPERSET_PLAN_STRUCTURE_PIVOT_SLICE, pivotParams).then(
+        async relevantJurisdictions => {
+          if (!relevantJurisdictions) {
+            return new Promise(reject => {
+              reject();
+            });
+          }
+          for (let i = 0; i < relevantJurisdictions.length; i += 1) {
+            const jurId = relevantJurisdictions[i].jurisdiction_id;
+            if (i) {
+              sqlFilterExpression += ' OR ';
```

Comment:
Why don't you use an `IN` (e.g. `jurisdiction_id IN (1,2,3)`) here?  It is supposed to be more efficient SQL-wise, and it would mean we don't need this loop here.